mods/page_revisions.js

```diff
@@ -211,6 +253,7 @@ PRS.prototype.getTitleRevision = function(restbase, req) {
         throw new Error("Invalid revision: " + rp.revision);
     }
     return revisionRequest
+    .then(self._checkRevReturn)
```

Comment:
I think you can make _checkRevReturn sync & move it to the callback. _checkRevReturn can throw the HTTPError if it doesn't check out.